File: 1455. Check If a Word Occurs As a Prefix of Any Word in a Sentence/solution.py
class Solution(object):
  def isPrefixOfWord(self, sentence, searchWord):
    """
    :type sentence: str
    :type searchWord: str
    :rtype: int
    """
    word = 1
    search = 0
    jump = False
    
    for letter in sentence:
      if jump and letter != " ":
        continue
      elif letter == " ":
        jump = False
        search = 0
        word += 1
        continue
      if letter == searchWord[search]:
        search += 1
      else:
        jump = True
        search = 0
      if search == len(searchWord):
        return word
      
    return -1

File: 1455. Check If a Word Occurs As a Prefix of Any Word in a Sentence/solution.cpp
class Solution {
public:
  int isPrefixOfWord(string sentence, string searchWord) {
    int word = 1;
    int search = 0;
    bool jump = false;

    for (int index = 0; index < sentence.size(); index++){
      char letter = sentence[index];
      if (jump && letter != ' '){
        continue;
      }
      else if (letter == ' '){
        jump = false;
        search = 0;
        word += 1;
        continue;
      }
      if (letter == searchWord[search]){
        search += 1;
      }
      else{
        jump = true;
        search = 0;
      }
      if (search == searchWord.size()){
        return word;
      }
    }

    return -1;
  }
};